tests/headers.test.ts

```diff
@@ -92,6 +92,26 @@ describe('using class', () => {
       });
 
     })
+
+    describe('gets fresh dynamic headers before each request', () => {
+      test('with request method', async () => {
+        const objectChangedThroughReference = { 'x-foo': 'old' }
+        const client = new GraphQLClient(ctx.url, { headers: () => objectChangedThroughReference });
```

Comment:
Note that I'm only testing the headers-as-object way, there's also `new Headers` and array. I didn't want to duplicate things that much.